src/plugins/intel_cpu/src/compiled_model.cpp

```diff
@@ -49,6 +49,8 @@ CompiledModel::~CompiledModel() {
         m_sub_compiled_models.clear();
         m_sub_memory_manager->_memorys_table.clear();
     }
+    auto streamsExecutor = std::dynamic_pointer_cast<ov::threading::IStreamsExecutor>(m_task_executor);
```

Comment:
shall we use `ov_type_ptr` instead?

Reply:
I don't quite understand what you mean, could you explain it in more detail? Thank you.

Reply:
more like @t-jankowski to be called

Reply:
It's [safe](https://github.com/openvinotoolkit/openvino/blob/e8d01dc46fd8c74feb0ae79ea28809c9acee1290/src/inference/src/dev/threading/istreams_executor.cpp#L23) to use `std` here. `IStreamsExecutor` and its derived classes do not provide OV RTTI so `ov::as_type_ptr` couldn't be used anyway.